src/Compatibility/Microsoft.DotNet.ApiCompat.Shared/RegexStringTransformer.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.ApiCompat
+{
+    /// <summary>
+    /// A regex string transformer that transforms an input string via a regex pattern to a replacement string.
```

Comment:
This isn't technically accurate, in reality what this is doing is to transform an input string against _**a set**_ of regex patterns and replacement strings, as opposed to just a single one.

Reply:
I wonder if we should instead have the pattern and replacement string be part of each left and right item themselves, in case you want to special case and only apply a single regex transform for a very specific left, but not the rest.

Reply:
I considered that but I chose the existing implementation for couple of reasons:
1. Passing in and parsing multiple transformation patterns for every left and right assembly is hard (think about the non msbuild usage).
2. In the msbuild world, in most cases this runs as part of a project's outer build which has consistent output paths for inner builds. In the console case, you most likely want to pass in a directory that contains a list of assemblies for which you wouldn't need a per assembly transformation pattern. I don't think that you would ever want to pass in assemblies that reside under completely different output paths and then check-in the generated suppression file.